### PR TITLE
209 new connector for numcosmo

### PR DESCRIPTION
**Pull request by vitenti**

### CODE REVIEW

File: firecrown/connector/numcosmo/numcosmo.py [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+        p_ml: Optional[Nc.PowspecML] = None,
+        p_mnl: Optional[Nc.PowspecMNL] = None,
+        dist: Nc.Distance,
+    ):
+        super().__init__(require_nonlinear_pk=require_nonlinear_pk)
+        self.p_ml = p_ml
+        self.p_mnl = p_mnl
+        self.dist = dist
+
+    def get_params_names(self):
```
Comment (marcpaterno) [inline firecrown/connector/numcosmo/numcosmo.py:45]: We should remove these 4 functions and make them non-virtual in the base class. We should also deprecate them in the base class, preparing to remove them from the public interface in firecrown 2.0

File: firecrown/likelihood/gauss_family/gauss_family.py [hunk 2]
```diff
@@ -42,35 +43,85 @@ def read(self, sacc_data: sacc.Sacc) -> None:
...
+        theory_vector_list: List[np.ndarray] = []
+        for stat in self.statistics:
+            theory = stat.compute_theory_vector(cosmo)
+            theory_vector_list.append(np.atleast_1d(theory))
+
+        return np.concatenate(theory_vector_list)
+
+    @final
+    def compute(self, cosmo: pyccl.Cosmology) -> Tuple[np.ndarray, np.ndarray]:
         """Calculate and return the chi-squared for the given cosmology."""
```
Comment (marcpaterno) [inline firecrown/likelihood/gauss_family/gauss_family.py:91]: The comment needs updating to reflect the new purpose of compute.

**Review by marcpaterno — CHANGES_REQUESTED** [review]
There are a few small change requests embedded in the comments above.

File: firecrown/connector/numcosmo/numcosmo.py (hunk 1 above)
Comment (vitenti) [reply firecrown/connector/numcosmo/numcosmo.py:45]: Done!

File: firecrown/connector/numcosmo/numcosmo.py [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+                "k": k,
+                "delta_matter:delta_matter": p_k,
+            }
+
+        if self.p_mnl:
+            p_mnl_spline = self.p_mnl.get_spline_2d(hi_cosmo)
+            z = np.array(  # pylint: disable-msg=invalid-name
+                np.array(p_mnl_spline.xv.dup_array())
+            )
+            k = np.array(np.array(p_mnl_spline.yv.dup_array()))
```
[the remark this answers is not in the document]
Comment (vitenti) [reply firecrown/connector/numcosmo/numcosmo.py:153]: Done!

File: firecrown/likelihood/gauss_family/gauss_family.py (hunk 2 above)
Comment (vitenti) [reply firecrown/likelihood/gauss_family/gauss_family.py:91]: Done!

**Review by marcpaterno — APPROVED** [review]

**Pull request merged**